dtfs-central-api/.eslintrc.js

```diff
@@ -90,5 +90,22 @@ module.exports = {
         '@typescript-eslint/no-explicit-any': 'off',
       },
     },
+    // Typescript test files
+    {
+      files: ['*.test.ts'],
+      extends: ['airbnb-base', 'plugin:@typescript-eslint/recommended-type-checked', 'plugin:prettier/recommended'],
+      plugins: ['@typescript-eslint'],
+      parser: '@typescript-eslint/parser',
+      parserOptions: {
+        ...baseParserOptions,
+        project: './tsconfig.eslint.json',
+        tsconfigRootDir: __dirname,
+      },
+      rules: {
+        ...baseRules,
+        '@typescript-eslint/unbound-method': 'off',
+        'no-param-reassign': 'off',
```

Comment:
The `no-param-reassign` was to remove an inline eslint disable, but considering it is just one place I'd rather keep it inline.

The `unbound-method` comes from the new `mockQueryRunner` I am passing around - each function defined on it (because I am using the `as unknown as QueryRunner` type assertion) is defined as a class method, meaning it is bound to whatever `this` is. Typescript eslint does not like this and prefers to use arrow functions where the context of `this` is not clear. An example of where this error appears is `expect(mockQueryRunner.connect).toHaveBeenCalled()` - `.connect` is considered to be unbound.

The solution to this is to temporarily modify `mock-query-runner.ts` to export each inidividual mock as well as the query runner. Considering that the PR after this will introduce a commonised transaction handler which removes the need for this, I will remove the `.eslintrc.js` changes and update `mock-query-runner.ts`